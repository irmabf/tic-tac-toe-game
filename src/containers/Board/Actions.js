import styled from 'styled-components'

const Actions = styled.div`
 margin: 0 auto
`

export default Actions
