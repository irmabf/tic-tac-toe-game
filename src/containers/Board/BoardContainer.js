import styled from 'styled-components'

const BoardContainer = styled.div`
 padding: 15px;
 background-color: #FFFFFF;
 border-radius: 4px;
`

export default BoardContainer
