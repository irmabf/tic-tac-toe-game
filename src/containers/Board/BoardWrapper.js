import styled from 'styled-components'

const BoardWrapper = styled.div`
	background-color: #FFFFFF;
	text-align: center;
	padding: 20px 20px 50px 20px;
	margin: 0 auto 40px auto;
	width: 100%;
	
	@media (min-width: 480px) {
		width: 450px;
		border-radius: 4px;
	}
`

export default BoardWrapper
