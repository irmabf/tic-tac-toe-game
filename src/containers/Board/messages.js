import { defineMessages } from 'react-intl'

export default defineMessages({
	heading: {
		id: 'tic-tac-toe.containers.Board.heading',
		defaultMessage: 'How do you want to play?'
	},
	onePlayer: {
		id: 'tic-tac-toe.containers.Board.onePlayer',
		defaultMessage: 'One Player'
	},
	twoPlayers: {
		id: 'tic-tac-toe.containers.Board.twoPlayers',
		defaultMessage: 'Two Playes'
	},
	doYouWantPlay: {
		id: 'tic-tac-toe.containers.Board.doYouWantPlay',
		defaultMessage: 'Do you want play as **{X}** or **{Y}**?'
	},
	x: {
		id: 'tic-tac-toe.containers.Board.x',
		defaultMessage: 'X'
	},
	y: {
		id: 'tic-tac-toe.containers.Board.y',
		defaultMessage: 'Y'
	},
	reset: {
		id: 'tic-tac-toe.containers.Board.reset',
		defaultMessage: 'Reset'
	},
	turn: {
		id: 'tic-tac-toe.containers.Board.turn',
		defaultMessage: 'It\'s **{who}** turn'
	},
	your: {
		id: 'tic-tac-toe.containers.Board.your',
		defaultMessage: 'Your'
	},
	computer: {
		id: 'tic-tac-toe.containers.Board.computer',
		defaultMessage: 'Computer'
	},
	theWinneIs: {
		id: 'tic-tac-toe.containers.Board.theWinneIs',
		defaultMessage: 'The winner is...{winner}!!!'
	},
	draw: {
		id: 'tic-tac-toe.containers.Board.draw',
		defaultMessage: 'It\'s a **Draw**'
	},
	you: {
		id: 'tic-tac-toe.containers.Board.you',
		defaultMessage: '**You**'
	}
})
