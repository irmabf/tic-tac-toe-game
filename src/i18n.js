/**
 * i18n.js
 *
 * This will setup the i18n language files and locale data for your app.
 *
 */
import { addLocaleData } from 'react-intl/lib/index.es'
import enLocaleData from 'react-intl/locale-data/en'
import deLocaleData from 'react-intl/locale-data/de'

import enTranslationMessages from './translations/en.json'
import deTranslationMessages from './translations/de.json'

addLocaleData(enLocaleData)
addLocaleData(deLocaleData)

export const appLocales = [
	'en',
	'de',
]

export const formatTranslationMessages = (locale, messages) => {
	const defaultFormattedMessages = locale !== 'en'
    ? formatTranslationMessages('en', enTranslationMessages)
    : {}
	return Object.keys(messages).reduce((formattedMessages, key) => {
		const formattedMessage = !messages[key] && locale !== 'en'
      ? defaultFormattedMessages[key]
      : messages[key]
		return Object.assign(formattedMessages, { [key]: formattedMessage })
	}, {})
}

export const translationMessages = {
	en: formatTranslationMessages('en', enTranslationMessages),
	de: formatTranslationMessages('de', deTranslationMessages),
}
