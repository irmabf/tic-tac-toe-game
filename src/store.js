import { applyMiddleware, combineReducers, createStore, compose } from 'redux'
import { persistStore, autoRehydrate } from 'redux-persist'
import { REHYDRATE } from 'redux-persist/constants'
import createActionBuffer from 'redux-action-buffer'
import { createLogger } from 'redux-logger'
import thunk from 'redux-thunk'

import app from './ducks/app'

const reducers = combineReducers({
	app
})

const store = createStore(reducers, compose(
	applyMiddleware(thunk, createLogger(), createActionBuffer(REHYDRATE)),
	autoRehydrate(),
	window.devToolsExtension ? window.devToolsExtension() : (f) => f
))

persistStore(store)

export default store
