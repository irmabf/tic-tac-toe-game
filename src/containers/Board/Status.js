import React, { Component } from 'preact-compat'
import PropTypes from 'prop-types'

import H2 from 'components/H2'
import FormattedMarkdownMessage from 'components/FormattedMarkdownMessage'
import messages from './messages'

class Status extends Component {

	getInfoMessage() {
		const { step, onePlayer, currentPlayer, firstPlayerSymbol, secondPlayerSymbol } = this.props
		let message
		switch (step) {
			case 4: {
				const onePlayerWinner =
					currentPlayer === firstPlayerSymbol ? messages.you.defaultMessage : messages.computer.defaultMessage
				const playerWinner =
					currentPlayer === firstPlayerSymbol ? firstPlayerSymbol : secondPlayerSymbol
				const winner = onePlayer ? onePlayerWinner : playerWinner
				message = (
					<FormattedMarkdownMessage
						{...messages.theWinneIs}
						values={{ winner: `<strong>${winner}</strong>` }}
					/>
				)
			}
				break
			case 5:
				message = (
					<FormattedMarkdownMessage {...messages.draw} />
				)
				break
			default:
				message = this.getTurn()
		}
		return message
	}

	getTurn() {
		const { onePlayer, currentPlayer, firstPlayerSymbol } = this.props
		return (
			onePlayer ? (
				<FormattedMarkdownMessage
					{...messages.turn}
					values={{ who:
						currentPlayer === firstPlayerSymbol ? messages.your.defaultMessage : messages.computer.defaultMessage
					}}
				/>
			) : (
				<FormattedMarkdownMessage
					{...messages.turn}
					values={{ who: currentPlayer }}
				/>
			)
		)
	}

	render() {
		return (
			<H2>{this.getInfoMessage()}</H2>
		)
	}
}

Status.propTypes = {
	step: PropTypes.number.isRequired,
	currentPlayer: PropTypes.string.isRequired,
	onePlayer: PropTypes.bool.isRequired,
	firstPlayerSymbol: PropTypes.string.isRequired,
	secondPlayerSymbol: PropTypes.string.isRequired
}

export default Status
