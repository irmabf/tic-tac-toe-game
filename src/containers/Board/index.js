import React, { Component } from 'react'
import { FormattedMessage } from 'react-intl'
import { connect } from 'react-redux'

import H2 from 'components/H2'
import Button from 'components/Button'
import { actions } from '../../ducks/game'
import BoardContainer from './BoardContainer'
import Actions from './Actions'
import messages from './messages'
import Game from '../Game'

class Board extends Component {

	setGameInfo() {
			const { onePlayer, setFirstPlayerSymbol, setOnePlayer } = this.props
			return (
				!onePlayer ? (
					<div>
						<H2>
							<FormattedMessage {...messages.heading} />
						</H2>
						<Actions>
							<Button onClick={() => setOnePlayer(true)}>
								<FormattedMessage {...messages.onePlayer} />
							</Button>
							<Button onClick={() => setOnePlayer(false)}>
								<FormattedMessage {...messages.twoPlayers} />
							</Button>
						</Actions>
					</div>
				) : (
					<div>
						<H2>
							<FormattedMessage
								{...messages.doYouWantPlay}
								values={{ X: <b>{messages.x.defaultMessage}</b>, Y: <b>{messages.y.defaultMessage}</b>}}
							/>
						</H2>
						<Actions>
							<Button onClick={() => setFirstPlayerSymbol('X')} xo>
								<FormattedMessage {...messages.x} />
							</Button>
							<Button onClick={() => setFirstPlayerSymbol('O')} xo>
								<FormattedMessage {...messages.y} />
							</Button>
						</Actions>
					</div>
				)
			)
	}

	render() {
		const { onePlayer, firstPlayerSymbol, squares } = this.props
		return (
			<BoardContainer>
				{onePlayer && firstPlayerSymbol  ? (<Game squares={squares} />) : (this.setGameInfo(onePlayer))}
			</BoardContainer>
		)
	}
}

const mapStateToProps = (state) => ({
	...state
})

const mapDispatchToProps = {
	setOnePlayer: actions.setOnePlayer,
	setFirstPlayerSymbol: actions.setFirstPlayerSymbol
}

export default connect(mapStateToProps, mapDispatchToProps)(Board)
