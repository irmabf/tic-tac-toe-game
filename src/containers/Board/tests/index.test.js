import React from 'react'
import { Provider } from 'react-redux'
import { mount } from 'enzyme'

import LanguageProvider from '../../../containers/LanguageProvider'
import { translationMessages } from '../../../i18n'
import store from '../../../store'
import Board from '../index'
import BoardWrapper from '../BoardWrapper'


describe('<Board />', () => {
	it('should render <BoardWrapper>', () => {
		const renderedComponent = mount(
			<Provider store={store}>
				<LanguageProvider messages={translationMessages}>
					<Board />
				</LanguageProvider>
			</Provider>
		)
		expect(renderedComponent.find(BoardWrapper).length).toEqual(1)
	})
})
