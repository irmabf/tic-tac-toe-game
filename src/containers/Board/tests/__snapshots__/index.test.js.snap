// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Board /> <Board /> renders correctly 1`] = `
<div
  className="BoardWrapper-brevKX gIZNep"
>
  <div>
    <h2
      className="H2-edqtgr NFMjd"
    >
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "How do you want to play?",
          }
        }
      />
    </h2>
    <a
      className="A-jrLLFi jpbFUk"
      href={undefined}
      onClick={[Function]}
    >
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "One Player",
          }
        }
      />
    </a>
    <a
      className="A-jrLLFi jpbFUk"
      href={undefined}
      onClick={[Function]}
    >
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "Two Playes",
          }
        }
      />
    </a>
  </div>
</div>
`;
